wasmer-wasi-c-api-example.c

```diff
@@ -0,0 +1,184 @@
+                                    .tag = WASM_FUNCTION,
+                                    .value.func = func };
+
+    // Define a memory import
+    const char *import_memory_name = "memory";
+    wasmer_byte_array import_memory_name_bytes = { .bytes = (const uint8_t *) import_memory_name,
+                                                   .bytes_len = strlen(import_memory_name) };
+    wasmer_import_t memory_import = { .module_name = module_name_bytes,
+                                      .import_name = import_memory_name_bytes,
+                                      .tag = WASM_MEMORY };
+    wasmer_memory_t *memory = NULL;
+    wasmer_limit_option_t max = { .has_some = true,
+                                  .some = 256 };
+    wasmer_limits_t descriptor = { .min = 256,
+                                   .max = max };
+
+    wasmer_result_t memory_result = wasmer_memory_new(&memory, descriptor);
+    if (memory_result != WASMER_OK)
+    {
+        print_wasmer_error();
+    }
+    memory_import.value.memory = memory;
+
+    // Define a global import
+    const char *import_global_name = "__memory_base";
+    wasmer_byte_array import_global_name_bytes = { .bytes = (const uint8_t *) import_global_name,
+                                                   .bytes_len = strlen(import_global_name) };
+    wasmer_import_t global_import = { .module_name = module_name_bytes,
+                                      .import_name = import_global_name_bytes,
+                                      .tag = WASM_GLOBAL };
+
+    wasmer_value_t val = { .tag = WASM_I32,
+                           .value.I32 = 1024 };
+    wasmer_global_t *global = wasmer_global_new(val, false);
+    global_import.value.global = global;
+
+    // Get the WASI base imports
+    const char *arg1_val = "wasi-example-program-name";
+    const char *arg2_val = "--help";
+    wasmer_byte_array args[2] = { { .bytes = (const uint8_t *) arg1_val,
+                                   .bytes_len = strlen(arg1_val) },
+                                 { .bytes = (const uint8_t *) arg2_val,
+                                   .bytes_len = strlen(arg2_val) } };
+    const int num_args = (sizeof args) / (sizeof args[0]);
+    
+    const char *env1_val = "COLOR_OUTPUT=1";
+    wasmer_byte_array envs[1] = { { .bytes = (const uint8_t *) env1_val,
+                                .bytes_len = strlen(env1_val) } };
+    const int num_envs = (sizeof envs) / (sizeof envs[0]);
+
+    wasmer_import_object_t *import_object =
+            wasmer_wasi_generate_import_object(args, num_args, envs, num_envs, NULL, 0, NULL, 0);
+    // Define an array containing our imports
+    wasmer_import_t extra_imports[] = {func_import, global_import, memory_import};
+
+    // Add our custom imports to the WASI ImportObject
+    wasmer_import_object_extend(import_object, extra_imports, 3);
+
+    // Read the wasm file bytes
+    FILE *file = fopen("wasm-wasi-sample-app/wasi_example.wasm", "r");
```

Comment:
I would rename `source.c` to `program.c`, and `wasi_example.wasm` to `program.wasm`, thoughts?